### PR TITLE
[android] Log and handle Android low memory events

### LINKED ISSUE #4648: Detect/handle low memory events on Android
Now we can't say if OM was killed due to a low-memory event. At first, it would be great to start logging these events:

https://developer.android.com/reference/android/content/ComponentCallbacks2#onTrimMemory(int)

https://developer.android.com/topic/performance/memory

At second, it may help to call Framework::MemoryWarning() from this event.

### PR DESCRIPTION
**Pull request by cyber-toad**
Fixes #4648

### CODE REVIEW

File: android/src/app/organicmaps/MwmActivity.java [hunk 1]
```diff
@@ -1857,4 +1857,15 @@ public void onPlacePageRequestToggleRouteSettings(@NonNull RoadType roadType)
...
+
+  @Override
+  public void onTrimMemory(int level)
+  {
+    super.onTrimMemory(level);
+    Logger.d(TAG, "trim memory, level = " + level);
+    if (level >= TRIM_MEMORY_RUNNING_LOW)
+    {
+      Framework.nativeMemoryWarning();
+    }
```
Comment (biodranik) [inline android/src/app/organicmaps/MwmActivity.java:1869]:
```suggestion
    if (level >= TRIM_MEMORY_RUNNING_LOW)
      Framework.nativeMemoryWarning();
```

File: android/jni/app/organicmaps/Framework.cpp [hunk 2]
```diff
@@ -1835,4 +1835,11 @@ Java_app_organicmaps_Framework_nativeHasPlacePageInfo(JNIEnv *, jclass)
 {
   return static_cast<jboolean>(frm()->HasPlacePageInfo());
 }
+
+JNIEXPORT void JNICALL
+Java_app_organicmaps_Framework_nativeMemoryWarning(JNIEnv *, jclass)
+{
+    return frm()->MemoryWarning();
+}
```
Comment (biodranik) [inline android/jni/app/organicmaps/Framework.cpp:1843]:
```suggestion
{
  return frm()->MemoryWarning();
}
```

**Review by biodranik — COMMENTED** [review]
Did you test it in low-memory conditions? What happens?

**Comment by cyber-toad** [comment]
> Did you test it in low-memory conditions? What happens?

When application goes to background this method is called and I was able to see that c++ method is called. I was not able to simulate other types of events for this method on my phone.

**Review by vng — APPROVED** [review]
Hm, yes, MemoryWarning() was unused in Android ..

**Review by biodranik — APPROVED** [review]
Thank you!

**Pull request merged**